craft_parts/executor/environment.py

```diff
@@ -88,7 +90,8 @@ def _basic_environment_for_part(part: Part, *, step_info: StepInfo) -> Dict[str,
 
     :return: A dictionary containing the built-in environment.
     """
-    part_environment: Dict[str, str] = _get_step_environment(step_info)
+    part_environment: Dict[str, str] = _get_global_environment(step_info.project_info)
+    part_environment.update(_get_step_environment(step_info))
```

Comment:
maybe update the part_environment dict with information from the application here as from the point of view of the user, it is all part of the same thing

Reply:
Updated to include global variables when getting the step environment. The separate call for global will be used later when expanding global variables.